test/unit/modules/test_sfp_debounce.py

```diff
@@ -0,0 +1,79 @@
+# test_sfp_debounce.py
+import unittest
+
+from modules.sfp_debounce import sfp_debounce
+from sflib import SpiderFoot
+from spiderfoot import SpiderFootEvent, SpiderFootTarget
+
+
+class TestModuledebounce(unittest.TestCase):
+    """
+    Test modules.sfp_debounce
+    """
+
+    default_options = {
+        '_debug': False,  # Debug
+        '__logging': True,  # Logging in general
+        '__outputfilter': None,  # Event types to filter from modules' output
+        '_useragent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:62.0) Gecko/20100101 Firefox/62.0',  # User-Agent to use for HTTP requests
+        '_dnsserver': '',  # Override the default resolver
+        '_fetchtimeout': 5,  # number of seconds before giving up on a fetch
+        '_internettlds': 'https://publicsuffix.org/list/effective_tld_names.dat',
+        '_internettlds_cache': 72,
+        '_genericusers': "abuse,admin,billing,compliance,devnull,dns,ftp,hostmaster,inoc,ispfeedback,ispsupport,list-request,list,maildaemon,marketing,noc,no-reply,noreply,null,peering,peering-notify,peering-request,phish,phishing,postmaster,privacy,registrar,registry,root,routing-registry,rr,sales,security,spam,support,sysadmin,tech,undisclosed-recipients,unsubscribe,usenet,uucp,webmaster,www",
+        '__version__': '3.3-DEV',
+        '__database': 'spiderfoot.test.db',  # note: test database file
+        '__modules__': None,  # List of modules. Will be set after start-up.
+        '_socks1type': '',
+        '_socks2addr': '',
+        '_socks3port': '',
+        '_socks4user': '',
+        '_socks5pwd': '',
+        '_torctlport': 9051,
+        '__logstdout': False
+    }
+
+    def test_opts(self):
+        module = sfp_debounce()
+        self.assertEqual(len(module.opts), len(module.optdescs))
+
+    def test_setup(self):
+        """
+        Test setup(self, sfc, userOpts=dict())
+        """
+        sf = SpiderFoot(self.default_options)
+
+        module = sfp_debounce()
+        module.setup(sf, dict())
+
+    def test_watchedEvents_should_return_list(self):
+        module = sfp_debounce()
+        self.assertIsInstance(module.watchedEvents(), list)
+
+    def test_producedEvents_should_return_list(self):
+        module = sfp_debounce()
+        self.assertIsInstance(module.producedEvents(), list)
+
+    def test_handleEvent(self):
```

Comment:
```suggestion
    @unittest.skip("todo")
    def test_handleEvent(self):
```

The tests are failing due to the code pattern used in the unit test.

```
test/unit/modules/test_sfp_debounce.py:77: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

modules/sfp_debounce.py:99: in handleEvent

    data = self.queryEmailAddr(eventData)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <modules.sfp_debounce.sfp_debounce object at 0x7f6df8370700>

qry = 'example data'

    def queryEmailAddr(self, qry):

        res = self.sf.fetchUrl(

            f"https://disposable.debounce.io?email={qry}",

>           timeout=self.opts['_fetchtimeout'],

            useragent="SpiderFoot"

        )

E       KeyError: '_fetchtimeout'

modules/sfp_debounce.py:71: KeyError
```

The easiest way to fix this, for now, is to mark the test to be skipped.

Reply:
@bcoles thanks! It removed the '_fetchtimeout' error.